### PR TITLE
added arm command selector

### PR DESCRIPTION
**Pull request by emilycxliu**
selector will choose the arm command based on the game piece type and target position.

### CODE REVIEW

**Review by weizhang-2706 — APPROVED** [review]

File: src/main/java/frc/robot/commands/ArmCommandSelector.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  ArmSetpoint m_ArmSetPoint;
+  boolean m_slowerAcceleration;
+  ArmCommand m_ArmCommand;
+  
+  /** Creates a new ArmCommand. */
+  public ArmCommandSelector(Supplier<RobotGamePieceState> robotState, 
+                            ArmPosition position,
+                            boolean slowerAcceleration ) {
+    //get the current robotState: from container
+    m_robotState = robotState.get();
```
Comment (ErikCald) [inline src/main/java/frc/robot/commands/ArmCommandSelector.java:26]: The robotState will change a lot during a match. Here you're getting the robot state when the command is first constructed and then only using that state. Instead, you will need to get the state in initialize() and then check which state it is.

File: src/main/java/frc/robot/commands/ArmCommandSelector.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    m_slowerAcceleration = slowerAcceleration;
+    // Use addRequirements() here to declare subsystem dependencies.
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    switch (m_robotState) {
+      case NoGamePiece: 
+        m_ArmSetPoint = ArmSetpoint.HOME_WITH_GAMEPIECE;
```
Comment (ErikCald) [inline src/main/java/frc/robot/commands/ArmCommandSelector.java:37]: The NoGamePiece arm setpoint should be ArmSetpoint.CONE_PICKUP instead.

**Review by ErikCald — CHANGES_REQUESTED** [review]
This command should handle the cases for TOP, MIDDLE, BOTTOM and for pickup as well. I think it would be better to add a specific ArmPosition for pickup than have pickup be the default option.

**Review by ErikCald — APPROVED** [review]
This can merge.

Note that the slowerAcceleration boolean was moved to inside the ArmSetpoint enum, but it hasn't been removed from the constructor of ArmCommand yet.

Soon, someone will need to remove that boolean from ArmCommandSelector and remove it from the constructor of ArmCommand.

**Pull request merged**